crates/wascc-provider/src/states/image_pull.rs

```diff
@@ -0,0 +1,30 @@
+use kubelet::state::{PodChangeRx, State, Transition};
+use kubelet::{
+    pod::{Phase, Pod},
+    state,
+};
+use log::error;
+
+use crate::{make_status, PodState};
+
+use super::image_pull_backoff::ImagePullBackoff;
+use super::volume_mount::VolumeMount;
+
+state!(
+    /// Kubelet is pulling container images.
+    ImagePull,
+    PodState,
+    VolumeMount,
+    ImagePullBackoff,
+    {
+        pod_state.run_context.modules = match pod_state.store.fetch_pod_modules(&pod).await {
+            Ok(modules) => modules,
+            Err(e) => {
+                error!("{:?}", e);
+                return Ok(Transition::Error(ImagePullBackoff));
+            }
+        };
+        Ok(Transition::Advance(VolumeMount))
+    },
+    { make_status(Phase::Pending, "ImagePull") }
```

Comment:
Note for other reviewers: This doesn't have any container status updating yet, so any ideas will be helpful. Most likely we can do something using the current pod state. 

However, we can possibly keep things using the current methodology. Only a few of the steps need to update container statuses (namely the init containers and starting steps), so perhaps it is ok to ask for a full status implementation